mriqc/workflows/functional/base.py

```diff
@@ -77,21 +78,41 @@ def fmri_qc_workflow(name='funcMRIQC'):
     mem_gb = config.workflow.biggest_file_gb
 
     dataset = config.workflow.inputs.get('bold', [])
+    full_files = []
+    for bold_path in dataset:
+        try:
+            bold_len = nb.load(bold_path).shape[3]
+        except nb.filebasedimages.ImageFileError:
+            bold_len = config.workflow.min_len_bold
+        except IndexError:    # shape has only 3 elements
+            bold_len = 0
+        if bold_len >= config.workflow.min_len_bold:
+            full_files.append(bold_path)
+        else:
+            config.loggers.workflow.warn(
+                f'Dismissing {bold_path} for processing: insufficient number of '
+                f'timepoints ({bold_len}) to execute the workflow.'
+            )
 
     message = BUILDING_WORKFLOW.format(
         modality='functional',
         detail=(
-            f'for {len(dataset)} BOLD runs.'
-            if len(dataset) > 2
+            f'for {len(full_files)} BOLD runs.'
+            if len(full_files) > 2
             else f"({' and '.join('<%s>' % v for v in dataset)})."
         ),
     )
     config.loggers.workflow.info(message)
 
+    if full_files != dataset:
+        config.loggers.workflow.info(
+            f'Skipping short runs: {sorted(set(dataset) - set(full_files))}'
+        )
```

Comment:
If the inputs config is not updated, this will fail when creating reports (tested locally):
```suggestion
    if set(dataset) - set(full_files):
        config.workflow.inputs['bold'] = full_files
```